src/Npgsql/NpgsqlBuffer.cs

```diff
@@ -618,5 +618,71 @@ struct BitConverterUnion
         }
 
         #endregion
+
+        #region Postgis
+        //  Postgis returns bytes in ndr representation.
+        internal int PostgisReadInt32()
```

Comment:
So if I understand correctly you have to check the first byte to know whether the data is in MSB or LSB, and then call the appropriate method on NpgsqlBuffer, right?

If this is the case, we can add methods to NpgsqlBuffer which accept a parameter specifying MSB or LSB (e.g. a `ByteOrder` enum). The methods can internally do whatever is necessary, this way your handler will be less complicated...